fixes #168, valueerror if non-square blocksize on convolve 2d

richpsharp opened the pull request:
This PR adds a check in convolve_2d that raises an exception with a helpful error message if the input rasters are in a row format. I also added a test case to cover this new functionality, and an update to the docstring and history.

richpsharp committed:
fixes #168, tests if the input blocksize has a value of 1 for any dimension and raises a helpful valueerror if so, test to cover, and udpate to history, and update to docstring

richpsharp commented:
@dcdenu4 I added you as a reviewer because of your "Interesting!" comment on https://github.com/natcap/pygeoprocessing/issues/168

dcdenu4 commented on src/pygeoprocessing/geoprocessing.py:
```diff
@@ -2572,15 +2576,24 @@ def convolve_2d(
+        if 1 in info_dict['block_size']:
+            raise ValueError(
+                f'{signal_path_band} has a row blocksize which can make this '
+                f'function run very slow, create a square blocksize using '
+                f'`warp_raster` or `align_and_resize_raster_stack` which '
+                f'creates square blocksizes by default')
```
Since I mentioned it in my other PR I'll mention it here for consistency. Some not needed `f`-strings here, but it doesn't really matter.

richpsharp replied:
Hi Doug! I remember you raised this issue in the past and I changed it to only have the f string on the line that needed it, but I did a PR with james later and he found it confusing that on a multiline concatenated string that some had f's on them and others didn't and I tend to agree with that. So knowing that the f replacement is not used on all the strings in the concatenation i still prefer it because it's a consistent style with the entire string itself. So if you're okay to let that slide then great, and if not lets chat about it more.

dcdenu4 replied:
It is more aesthetically pleasing and I'm ultimately happy to let it slide. However, personally when I see a string start with an `f` I expect there to be a replacement happening in that string and when I can't find it, I get confused. To me, the f-string is misleading if there is nothing being replaced and I prefer it to be used with intent over aesthetics. But, pretty low stakes!

dcdenu4 commented on tests/test_geoprocessing.py:
```diff
@@ -4226,3 +4227,43 @@ def test_empty_vector_extent_creation(self):
+        pixel_size = (1, -1)
+        origin = (0, 0)
+
+        # make row block (can't do a column block because geotiffs force a
+        # larger blocksize in that case)
```
Interesting

dcdenu4 reviewed (commented):
Looks good to me! I just had a comment on some non needed f-strings... Sorry if that's over the top tedious.

richpsharp commented:
> Looks good to me! I just had a comment on some non needed f-strings... Sorry if that's over the top tedious.

I commented on your f-string comment! Lemme know what you think!

dcdenu4 reviewed (changes requested):
Hey @richpsharp , I added my perspective on the f-string hot debate. I'd approve and merge but it looks like there is a merge conflict in History from my previous merge!

richpsharp committed:
#168 history conflict

richpsharp commented:
> Hey @richpsharp , I added my perspective on the f-string hot debate. I'd approve and merge but it looks like there is a merge conflict in History from my previous merge!

Hi @dcdenu4, I fixed the merge. In the future if it's just a history conflict and you're comfortable with it, you can still approve and I can do the merge once i fix the history.rst conflict. Anyway, back to you again for a approve/merge!

richpsharp committed:
#168 history rst conflict

richpsharp committed:
#168 fixing an old merge conflict note

dcdenu4 reviewed (approved):
Approved
